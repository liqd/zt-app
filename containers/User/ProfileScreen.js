import React, { useEffect,useState } from 'react'
import { View } from 'react-native'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import AsyncStorage from '@react-native-async-storage/async-storage'
import { Button } from '@rneui/base'

import API from '../../BaseApi'
import { AvatarCircle } from '../../components/ButtonAvatar'
import { ButtonSignOut } from '../../components/ButtonSignOut'
import { Header } from '../../components/Header'
import { TextSourceSans } from '../../components/TextSourceSans'

import { styles } from './ProfileScreen.styles'

export const ProfileScreen = (props) => {
  const toSettingsOverview = () => {
    props.navigation.navigate('SettingsOverview', {
      userId: userId,
      userName: user.username,
      userImage: user._avatar})
  }

  const {userName, userImage} = props.route.params

  const settingsIcon = <IconSLI name='settings' size={30} />
  const rightHeaderButton = (
    <Button
      icon={settingsIcon}
      type='clear'
      onPress={toSettingsOverview}
    />)
  const {userId} = props.route.params
  const [user, setUser] = useState()

  const fetchUser = () => {
    return AsyncStorage.getItem('authToken')
      .then((token) => API.getUser(userId, token))
      .then((response) => {
        if(response.statusCode === 200) {
          setUser(response.data)
        } else {
          return Promise.reject(new Error('fetchUser returned ' + response.statusCode))
        }
      })
      .catch(error => console.warn(error))
  }

  useEffect(() => {
    fetchUser()
  }, [])

  return (
    <View style={styles.container}>
      <Header
        navigation={props.navigation}
        rightButton={user && user.is_self && rightHeaderButton}
      />
      <View style={styles.containerInner}>
        <AvatarCircle
          imgSource={userImage ? {uri: userImage} :user && {uri: user._avatar}}
          avatarStyles={styles.avatarStyles}
        />
        <TextSourceSans>
          {userName ? userName : user && user.username}
        </TextSourceSans>
      </View>
      {user && user.is_self &&
        <ButtonSignOut />
      }
    </View>
  )
}
