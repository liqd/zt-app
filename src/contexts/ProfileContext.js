import React, { createContext, useState } from 'react'

export const ProfileContext = createContext()

export const ProfileProvider = (props) => {
  const [profileContext, setProfileContext] = useState()

  return (
    <ProfileContext.Provider value={[profileContext, setProfileContext]}>
      {props.children}
    </ProfileContext.Provider>
  )
}
