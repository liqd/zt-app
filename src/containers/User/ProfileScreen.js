import React, { useContext } from 'react'
import { View } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import IconSLI from 'react-native-vector-icons/SimpleLineIcons'
import { Button } from '@rneui/base'

import { AvatarCircle } from '../../components/ButtonAvatar'
import { ButtonSignOut } from '../../components/ButtonSignOut'
import { Header } from '../../components/Header'
import { TextSourceSans } from '../../components/TextSourceSans'
import { ProfileContext } from '../../contexts/ProfileContext'
import { useUser } from '../../hooks/User'

import { styles } from './ProfileScreen.styles'

export const ProfileScreen = (props) => {
  const toSettingsOverview = () => {
    props.navigation.navigate('SettingsOverview')
  }

  const [profileContext] = useContext(ProfileContext)

  const settingsIcon = <IconSLI name='settings' size={30} />
  const rightHeaderButton = (
    <Button
      icon={settingsIcon}
      type='clear'
      onPress={toSettingsOverview}
    />)

  const user = useUser(profileContext?.userId)

  return (
    <SafeAreaView
      style={styles.safeAreaContainer}
    >
      <View style={styles.container}>
        <Header
          navigation={props.navigation}
          rightButton={user && user.is_self && rightHeaderButton}
        />
        <View style={styles.containerInner}>
          <AvatarCircle
            imgSource={{uri:
              profileContext?.userImage?.uri ||
              profileContext?.userImageFallback?.uri
            }}
            avatarStyles={styles.avatarStyles}
          />
          <TextSourceSans>
            {profileContext?.userName}
          </TextSourceSans>
        </View>
        {user && user.is_self &&
        <ButtonSignOut />
        }
      </View>
    </SafeAreaView>
  )
}
