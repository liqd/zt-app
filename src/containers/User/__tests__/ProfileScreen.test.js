import React from 'react'
import { render } from '@testing-library/react-native'

import '@testing-library/jest-native/extend-expect'

import { ProfileContext } from '../../../contexts/ProfileContext'
import { testUser } from '../../../tests/TestData'
import { ProfileScreen } from '../ProfileScreen.js'

const mockUser = testUser

jest.mock('../../../hooks/User', () => ({
  __esModule: true,
  useUser: () => mockUser
}))

test('Test ProfileScreen Snapshot', async () => {
  const route = { params: { userId: 1 } }
  const { findByText, toJSON } = render(
    <ProfileContext.Provider value={[{userName: 'testname'}]}>
      <ProfileScreen route={route} />
    </ProfileContext.Provider>
  )
  await findByText(/testname/)
  expect(toJSON()).toMatchSnapshot()
})
