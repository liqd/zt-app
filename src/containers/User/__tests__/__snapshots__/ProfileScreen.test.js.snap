// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfileScreen Snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 20,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 2,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 2,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginVertical": 0,
                  "padding": 8,
                  "paddingHorizontal": 10,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 22,
                    },
                    undefined,
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
              <Text
                style={
                  Object {
                    "color": "#1a1a1a",
                    "fontSize": 19.125,
                    "marginHorizontal": 5,
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Back
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 2,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#2089dc",
                  "borderRadius": 2,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 8,
                  "paddingHorizontal": 10,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 30,
                    },
                    undefined,
                    Object {
                      "fontFamily": "simple-line-icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Image
        accessibilityIgnoresInvertColors={true}
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Array [
            Object {
              "backgroundColor": "#2d40cc",
              "borderRadius": 100,
              "height": 40,
              "width": 40,
            },
            Object {
              "height": 112,
              "marginBottom": 20,
              "marginTop": 100,
              "width": 112,
            },
          ]
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "SourceSansPro_400Regular",
            "fontSize": 18,
          }
        }
      >
        testname
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 2,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#1a1a1a",
              "borderColor": "#2089dc",
              "borderRadius": 0,
              "borderWidth": 0,
              "flexDirection": "row",
              "fontSize": 18,
              "height": 70,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Sign out
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
